src/DynamicContent/Strings.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace WPML\PB\Elementor\DynamicContent;
+
+use WPML_PB_String;
+
+class Strings {
+
+	const SETTINGS_REGEX        = '/settings="(.*?(?="]))/';
+	const NAME_PREFIX           = 'dynamic';
+	const DELIMITER             = '-';
+	const TRANSLATABLE_SETTINGS = [
+		'before',
+		'after',
+		'fallback',
+	];
+
+	/**
+	 * Remove the strings overwritten with dynamic content
+	 * and add the extra strings "before", "after" and "fallback".
+	 *
+	 * @param WPML_PB_String[] $strings
+	 * @param string            $nodeId
+	 * @param array             $element
+	 *
+	 * @return WPML_PB_String[]
+	 */
+	public static function filter( array $strings, $nodeId, array $element ) {
+		if ( ! isset( $element['settings']['__dynamic__'] ) ) {
+			return $strings;
+		}
+
+		$dynamicFields = wpml_collect( $element['settings']['__dynamic__'] );
+
+		$isStaticStringOfDynamicField = function( WPML_PB_String $string ) use ( $dynamicFields ) {
```

Comment:
@strategio Unless you have a real need to target master I would target develop and use the FP functions like curry and partial to make the code clearer.

```php
$isStaticStringOfDynamicField = curryN( 2, function( $dynamicFields, WPML_PB_String $string ) {
...
} );

...
->reject( $isStaticStringOfDynamicField( $dynamicFields )
....
```

Reply:
@brucepearson, we decided to target `master` as much as we can because we have a lot of compatibility fixes to deliver. We might even request a compatibility specific update for Core (from `master`).

As soon as the FP library will be in `master`, we'll start using it.